File: app/src/main/java/com/candy/capture/customview/AudioIndicator.java
package com.candy.capture.customview;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.view.View;

import com.candy.commonlibrary.utils.DensityUtil;

/**
 * Created by zhanghq on 2016/11/15.
 */

public class AudioIndicator extends View {

    private Paint mPaint;

    private int mWidth;
    private float mScale = 1;

    public AudioIndicator(Context context) {
        super(context);

        mWidth = DensityUtil.dip2px(context, 5);
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG | Paint.DITHER_FLAG);
        mPaint.setStyle(Paint.Style.STROKE);
        mPaint.setStrokeCap(Paint.Cap.ROUND);
        mPaint.setStrokeWidth(mWidth);
    }

    public void setColor(int color) {
        mPaint.setColor(color);
    }

    public void setScale(float scale) {
        mScale = scale;
        requestLayout();
        invalidate();
    }

    public float getScale() {
        return mScale;
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int width = MeasureSpec.makeMeasureSpec(mWidth, MeasureSpec.EXACTLY);
        int height = MeasureSpec.makeMeasureSpec((int) (mWidth * mScale), MeasureSpec.EXACTLY);
        super.onMeasure(width, height);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        if (mScale <= 1) {
            canvas.drawPoint(getWidth() / 2, getHeight() / 2, mPaint);
        } else {
            canvas.drawLine(getWidth() / 2, mWidth / 2, getWidth() / 2, getHeight() - mWidth / 2, mPaint);
        }
    }
}

File: app/src/main/java/com/candy/capture/customview/AudioIndicator.kt
package com.candy.capture.customview

import android.content.Context
import android.graphics.Canvas
import android.graphics.Paint
import android.view.View
import com.candy.commonlibrary.utils.DensityUtil

/**
 * @Description
 * @Author zhanghaiqiang
 * @Date 2020/9/30 11:29
 */
class AudioIndicator(context: Context): View(context) {
    private var mPaint: Paint

    private var mWidth = 0
    private var mScale = 1f

    init {
        mWidth = DensityUtil.dip2px(context, 5f)
        mPaint = Paint(Paint.ANTI_ALIAS_FLAG or Paint.DITHER_FLAG)
        mPaint.style = Paint.Style.STROKE
        mPaint.strokeCap = Paint.Cap.ROUND
        mPaint.strokeWidth = mWidth.toFloat()
    }

    fun setColor(color: Int) {
        mPaint.color = color
    }

    fun setScale(scale: Float) {
        mScale = scale
        requestLayout()
        invalidate()
    }

    fun getScale(): Float {
        return mScale
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val width = MeasureSpec.makeMeasureSpec(mWidth, MeasureSpec.EXACTLY)
        val height = MeasureSpec.makeMeasureSpec((mWidth * mScale).toInt(), MeasureSpec.EXACTLY)
        super.onMeasure(width, height)
    }

    override fun onDraw(canvas: Canvas) {
        if (mScale <= 1) {
            canvas.drawPoint(width / 2.toFloat(), height / 2.toFloat(), mPaint)
        } else {
            canvas.drawLine(width / 2.toFloat(), mWidth / 2.toFloat(), width / 2.toFloat(), height - mWidth / 2.toFloat(), mPaint)
        }
    }
}